### ISSUE TITLE
Exception in tests give risky test as result

### ISSUE THREAD

**Issue by bobmulder** [cakephp/cakephp#7974]
Hi guys,

I created a controller-test that should get an exception. I asserted on this, and my test worked great, however, PHPUnit marks my test as risky. Using `verbose` gave me the following information:

> Test code or tested code did not (only) close its own output buffers

Googling on this brought me to this: https://github.com/Codeception/Codeception/issues/1262

(intersting comment: https://github.com/Codeception/Codeception/issues/1262#issuecomment-52419252)

I'm not an expert, but I guess Cakephp needs a small change to close the output buffer on an exception. Anybody knows how to make this work?

**Comment by markstory** [comment]
Are you running into this issue with `testAction()`? How could someone reproduce the issue you're seeing?

**Comment by dereuromark** [comment]
I think I reported this back in 3.0 RC days. Had the same issue back then.

**Comment by xavier83ar** [comment]
I've found a similar issue trying to test a controller action that should returns a file. After a little research I think that the problem is [here](https://github.com/cakephp/cakephp/blob/master/src/Network/Response.php#L1565). 
It seems that in this line cakephp is closing buffers that it should not, I'm not sure about that, but if I remove that line, test runs as expected. 

It seems that when raising the exception, PhpUnit checks the value of `ob_get_level()` before run a test and then recheck after run it, and those values doesn't match, due to the call to `ob_end_clean()`.

To reproduce the issue (at least the one that I've found), just create a method that returns a file and test it.

```php
class ExamplesController extends AppController 
{
    public function getFile() 
    {
        $this->response->file('/some/file.ext');
        return $this->response;
    }
}
```

```php
class ExamplesControllerTest extends IntegrationTestCase
{
    public function testGetFile()
    {
        $this->get('/examples/get-file');
        $this->assertResponseOk();
    }
}
```

This code has not been tested but I think it should help, or at least I hope so.
